File: ruby/hamming/hamming.rb
class Hamming
  VERSION = 1

  def self.compute(a, b)
    raise ArgumentError, "The DNA strands entered are of unequal length" unless a.size == b.size 
    (0...a.size).reduce(0) { |sum, i| a[i] != b[i] ? sum + 1 : sum }
  end
end

File: csharp/hamming/Hamming.cs
public class Hamming
{
  public static int Compute(string x, string y)
  {
    int diff = 0;
    for (int i = 0; i < x.Length; i++)
    {
      diff += x[i] != y[i] ? 1 : 0;
    }
    return diff;
  }
}